src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/KeyValuePairConverter.cs

```diff
@@ -91,28 +88,16 @@ internal override void Initialize(JsonSerializerOptions options, JsonTypeInfo? j
             return true;
         }
 
-        protected override void EndRead(ref ReadStack state)
```

Comment:
Can you check if this `EndRead` method is being overriden elsewhere? If not, it might make sense to remove the base method altogether.